devicetypes/smartthings/zwave-lock.src/zwave-lock.groovy

```diff
@@ -1588,7 +2741,73 @@ private def getChangeType(lockCodes, codeID) {
 }
 
 /**
- * Method to obtain status for descriptuion based on change type
+ * Checks if a schedule change type is set or update
+ *
+ * @param codeID The code slot number
+ *
+ * @param scheduleType The schedule type to check for
+ *
+ * @return "set" or "update" basis the presence of the code id in the lockCodesData map
+ */
+private def getScheduleChangeType(codeID, scheduleType) {
+	def changeType = "set"
+	def lockCodesData = device.currentValue("lockCodesData")
+	if (lockCodesData) {
+		lockCodesData = parseJson(lockCodesData)
+		def data = lockCodesData[codeID.toString()]
+		if (data) {
+			def scheduleData = data.scheduleInfo?.scheduleData
+			if (scheduleType == SCHEDULE_TYPE_DAILY_REPEATING || scheduleType == SCHEDULE_TYPE_WEEK_DAY) {
+				if (scheduleData && scheduleData.startTime && scheduleData.endTime && scheduleData.weekDayBitMask) {
+					changeType = "changed"
+				}
+			} else if (scheduleType == SCHEDULE_TYPE_YEAR_DAY) {
+				if (scheduleData && scheduleData.startDate && scheduleData.startTime && scheduleData.stopDate && scheduleData.stopTime) {
+					changeType = "changed"
+				}
+			}
+		}
+	}
+	changeType
+}
+
+/**
+ * Checks if schedule scanning is in progress
+ *
+ * @param codeID The code slot number
+ *
+ * @return true if scanning is in progress, else false
+ */
+private def isScheduleScanningInProgress(codeID) {
+	def codesToScanSchedules = state.codesToScanSchedules
+	if (codesToScanSchedules && codeID.toInteger() in codesToScanSchedules) {
```

Comment:
`return (state.codesToScanSchedules && (codeID.toInteger() in state.codesToScanSchedules))`